src/main/java/nl/tudelft/ti2206/group9/level/Track.java

```diff
@@ -77,10 +97,12 @@ public Track(final Random generator) {
     @SuppressWarnings("restriction")
     public final void moveTrack(final double dist) {
         synchronized (this) {
-            for (final AbstractEntity entity : entities) {
-                if (!(entity instanceof Player)) {
-                    moveEntity(entity, -dist);
+            for (int i = 0; i < entities.size(); i++) {
+                if (i == player) {
+                    continue;
                 }
+                AbstractEntity entity = entities.get(i);
+                moveEntity(entity, -dist);
```

Comment:
Performance-wise, it is not handy to assign new objects in a for-loop.
Suggestion: `moveEntity(entities.get(i), -dist);`

<review,logic